download_posts.py

```diff
@@ -1,25 +1,28 @@
 #!/usr/bin/python3
 
 import json
-import os 
-import requests 
+import os
+import requests
 from sys import exit as sysexit
 import xml.etree.ElementTree as ET
+from datetime import datetime, timedelta
+from dateutil.relativedelta import relativedelta
 
-# At time of script import, request range of years. We'll add 1 to the end year automatically.
-try:
-    startYear = int(input("Enter first year you want to export: "))
-except Exception as e:
-    print(f"\nError with first year entered. Error: {e}. Exiting...")
-    sysexit(1)
+DATE_FORMAT = '%Y%m%d'
 
-try:
-    endYear = int(input("Enter last year you want to export: "))
-except Exception as e:
-    print(f"\nError with last year entered. Error: {e}. Exiting...")
-    sysexit(1)
+def get_date_input(prompt):
+    try:
+        start_date = int(input("Enter start date in YYYYMMDD format: "))
+    except Exception as e:
+        print(f"\nError: {e}. Exiting...")
```

Comment:
Updating text because it’s not only year anymore is fair. Dropping the helpful information about the actual cause of the error is not nice for the user, however.